From mboxrd@z Thu Jan  1 00:00:00 1970
Received: from mail-oi1-f173.google.com (mail-oi1-f173.google.com [209.85.167.173])
	(using TLSv1.2 with cipher ECDHE-RSA-AES128-GCM-SHA256 (128/128 bits))
	(No client certificate requested)
	by smtp.subspace.kernel.org (Postfix) with ESMTPS id C4CD91F419A
	for <linux-wireless@vger.kernel.org>; Thu, 19 Feb 2026 00:40:31 +0000 (UTC)
Authentication-Results: smtp.subspace.kernel.org; arc=none smtp.client-ip=209.85.167.173
ARC-Seal:i=1; a=rsa-sha256; d=subspace.kernel.org; s=arc-20240116;
	t=1771461632; cv=none; b=MlWHdMapOCb0l+wYHd0JkE+8kFk3yednPjkkEcTs6HyuUdX/EqwzWCZJ8PYrakjqmN4czf+Ju9zGOpUdEgK+NLKq0Du07hIm5Wni41lQ3scS4VPEGZcRzdbA2d6ulGFDNzxVvWtUI2fWnDmS4q/25iSgUll5r7a/LgJLIBVICMs=
ARC-Message-Signature:i=1; a=rsa-sha256; d=subspace.kernel.org;
	s=arc-20240116; t=1771461632; c=relaxed/simple;
	bh=Eg4P+kcK6gJysYUwKafyTXzuMTJyGhEkGVGJu7nCeX4=;
	h=From:To:Cc:Subject:Date:Message-ID:In-Reply-To:References:
	 MIME-Version; b=Hm3ceU+fhlnNcowwxVIZhxN8P4Uh2nfAHeH7izHMa91XITBSM4HuTJli6z5PdMDRDA1W1Vg/MRn36ZIisK0QkoY5Gup7XbW2wrrMpJcUqCymOOJ8dj2LprJaA9qtKcQnTjQYblZJYABC4ZahfLuFxvht50vNck1biKiqIHUcN+g=
ARC-Authentication-Results:i=1; smtp.subspace.kernel.org; dmarc=fail (p=quarantine dis=none) header.from=kernel.org; spf=pass smtp.mailfrom=gmail.com; arc=none smtp.client-ip=209.85.167.173
Authentication-Results: smtp.subspace.kernel.org; dmarc=fail (p=quarantine dis=none) header.from=kernel.org
Authentication-Results: smtp.subspace.kernel.org; spf=pass smtp.mailfrom=gmail.com
Received: by mail-oi1-f173.google.com with SMTP id 5614622812f47-45f04f1348cso274505b6e.1
        for <linux-wireless@vger.kernel.org>; Wed, 18 Feb 2026 16:40:31 -0800 (PST)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20230601; t=1771461631; x=1772066431;
        h=content-transfer-encoding:mime-version:references:in-reply-to
         :message-id:date:subject:cc:to:from:x-gm-gg:x-gm-message-state:from
         :to:cc:subject:date:message-id:reply-to;
        bh=2KVGtyL16XjDOB5+IDsHdTUZ/GMHT4lDhgpqrINKBkU=;
        b=B5TNdEfq3adV8sdujLjahCqlqteqdoK37UFA+GEJRDqzDtLixGNx6QKZw4Pt2BEDIR
         V00FdFpVswUhTeMVixYcewat0Y7FUIs3zOzCtWHe9Oc0D8eMkaqDxwmkCKsMVGk+SyT7
         HwGckF5ik7KCXB+hwO930F1I5pU17xkq2BfuwynYVHLiLw4IX386rPGZIUGelRksKqJ7
         iiXSoREFEd0iiPMZGc1fK2KRqD0RhewhVOTeGvC5tObtgko6MRusGzzqo1iUuzmx6h8/
         9tvpmhlZk7W3yq66XwXMGgeLuS/Uh/gCH5zACIfBxKxFpZl8CEWPhazvjhhSzRfye4BC
         Uk0w==
X-Gm-Message-State: AOJu0Yx5GB00h6F1himDD8HEwUl3MnM+TbuHnjU5nUHPUwCwyZ6m/E7h
	/E4eivGZSZ1xbN6K14fumOUnTaLVP0hnuVju9w8siuXz3nUuox1cObzZ
X-Gm-Gg: AZuq6aK1t3kTdFz70k7zvm60IWr7o9C9BXc95/G8I3LcPCsKQY+U7XvEhMhiOExuOcg
	b1c53+9BrW34OEcZiXxmtsRSihg/KyDplmXV/99nHsJc/ILTpSgPAx13OTbi9VQJkWLMpDGFWFg
	nsnOdfy0QkVMJRYYnmQF+t0kX9RUTzSlnrqvHkzF5z70ammevLCpNEL4ngiVfUf3mGHo6hgn9fE
	Zz0KsazVi19Rlt6Mr6ybvA2KX/3KqFMyOxb1M6wt2vj/tELAoiPMwpC+jv5NQW2w8+DAx1tbmPB
	s4iVCZVRHyE4e6szB6NUYSMJjp+PILP6N6k4oCvo537dHsrWT44wshivqaRvHEOjtMBJgQc4Iea
	exNXaj0FeeD+nn5MUCYwHaWz1VJpZWM1n+scTOCSlC0qOBMJkeY4tCziclnZYnL1a7EjabhIGQ5
	FdR0zft+bf0Qe980qKPC0NvPxTTaKZL44RMh9LKAkGAPYoZg4=
X-Received: by 2002:a05:6808:2445:b0:450:5e3a:6f1a with SMTP id 5614622812f47-463b3f655e2mr8759789b6e.20.1771461630814;
        Wed, 18 Feb 2026 16:40:30 -0800 (PST)
Received: from sean-HP-EliteBook-830-G6.lan ([207.191.35.252])
        by smtp.gmail.com with ESMTPSA id 586e51a60fabf-40eaee45d12sm23477659fac.1.2026.02.18.16.40.29
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Wed, 18 Feb 2026 16:40:29 -0800 (PST)
From: sean.wang@kernel.org
To: nbd@nbd.name,
	lorenzo.bianconi@redhat.com
Cc: linux-wireless@vger.kernel.org,
	linux-mediatek@lists.infradead.org,
	Sean Wang <sean.wang@mediatek.com>
Subject: [PATCH 11/11] wifi: mt76: mt7921: add MT7902 SDIO device support
Date: Wed, 18 Feb 2026 18:40:07 -0600
Message-ID: <20260219004007.19733-11-sean.wang@kernel.org>
X-Mailer: git-send-email 2.43.0
In-Reply-To: <20260219004007.19733-1-sean.wang@kernel.org>
References: <20260219004007.19733-1-sean.wang@kernel.org>
Precedence: bulk
X-Mailing-List: linux-wireless@vger.kernel.org
List-Id: <linux-wireless.vger.kernel.org>
List-Subscribe: <mailto:linux-wireless+subscribe@vger.kernel.org>
List-Unsubscribe: <mailto:linux-wireless+unsubscribe@vger.kernel.org>
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit

From: Sean Wang <sean.wang@mediatek.com>

Register the MT7902 SDIO device ID in the mt7921 driver and add its
corresponding firmware and ROM patch names.

Signed-off-by: Sean Wang <sean.wang@mediatek.com>
---
 drivers/net/wireless/mediatek/mt76/mt7921/sdio.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/net/wireless/mediatek/mt76/mt7921/sdio.c b/drivers/net/wireless/mediatek/mt76/mt7921/sdio.c
index 3421e53dc948..9150f185716c 100644
--- a/drivers/net/wireless/mediatek/mt76/mt7921/sdio.c
+++ b/drivers/net/wireless/mediatek/mt76/mt7921/sdio.c
@@ -19,6 +19,8 @@
 static const struct sdio_device_id mt7921s_table[] = {
 	{ SDIO_DEVICE(SDIO_VENDOR_ID_MEDIATEK, 0x7901),
 		.driver_data = (kernel_ulong_t)MT7921_FIRMWARE_WM },
+	{ SDIO_DEVICE(SDIO_VENDOR_ID_MEDIATEK, 0x7902),
+		.driver_data = (kernel_ulong_t)MT7902_FIRMWARE_WM },
 	{ }	/* Terminating entry */
 };
 
@@ -317,6 +319,8 @@ static int mt7921s_resume(struct device *__dev)
 MODULE_DEVICE_TABLE(sdio, mt7921s_table);
 MODULE_FIRMWARE(MT7921_FIRMWARE_WM);
 MODULE_FIRMWARE(MT7921_ROM_PATCH);
+MODULE_FIRMWARE(MT7902_FIRMWARE_WM);
+MODULE_FIRMWARE(MT7902_ROM_PATCH);
 
 static DEFINE_SIMPLE_DEV_PM_OPS(mt7921s_pm_ops, mt7921s_suspend, mt7921s_resume);
 
-- 
2.43.0


